### PR TITLE
Better error reporting and clean up of libdeno namespace

### PR DESCRIPTION
**Pull request by ry**
This is mostly the changes that were in #452 but it was getting to be quite complicated, so I'm doing this in a separate PR.

### CODE REVIEW

**Comment by ry** [comment]
@piscisaureus @kitsonk PTAL

File: src/deno_dir.rs [hunk 1]
```diff
@@ -165,12 +165,27 @@ impl DenoDir {
 
     let j: Url =
       if containing_file == "." || Path::new(module_specifier).is_absolute() {
-        Url::from_file_path(module_specifier).unwrap()
+        let r = Url::from_file_path(module_specifier);
+        // TODO(ry) Properly handle error.
+        if r.is_err() {
+          error!("Url::from_file_path error {}", module_specifier);
```
Comment (piscisaureus) [inline src/deno_dir.rs]: I'm not sure what's the practical difference with the situation prior to this patch.
But it's not worse either. 🤷‍♂️

File: tools/format.py [hunk 2]
```diff
@@ -29,7 +29,6 @@
 rustfmt_extra_args = []
 if 'RUSTFMT_FLAGS' in os.environ:
```
Comment (piscisaureus) [inline tools/format.py]: FYI I don't `RUSTFMT_FLAGS` any more. 
You can remove or I will do it myself.
(^-- this sounds unintentionally threatening)

**Comment by piscisaureus** [comment]
I don't see any "clean up of libdeno namespace". This PR complete?

File: src/deno_dir.rs (hunk 1 above)
Comment (ry) [reply src/deno_dir.rs]: It sometimes would fail on windows due to not having "C:/" ... and it wouldn't print anything.

File: js/globals.ts [hunk 3]
```diff
@@ -28,10 +20,21 @@ declare global {
...
+type MessageCallback = (msg: Uint8Array) => void;
+interface Libdeno {
+  recv(cb: MessageCallback): void;
+  send(msg: ArrayBufferView): null | Uint8Array;
+  print(x: string): void;
+  mainSource: string;
+  mainSourceMap: RawSourceMap;
+}
+export const libdeno = window["libdeno"] as Libdeno;
+window["libdeno"] = null;
```
Comment (piscisaureus) [inline js/globals.ts]: delete window.libdeno ?

File: src/binding.cc [hunk 4]
```diff
@@ -295,17 +296,34 @@ void InitializeContext(v8::Isolate* isolate, v8::Local<v8::Context> context,
...
+      v8::ScriptOrigin origin(v8_str("set_source_map.js"));
+      std::string source_map_parens = "(" + *source_map + ")";
+      auto source_map_v8_str = deno::v8_str(source_map_parens.c_str());
+      auto script = v8::Script::Compile(context, source_map_v8_str, &origin);
+      if (script.IsEmpty()) {
+        DCHECK(try_catch.HasCaught());
+        HandleException(context, try_catch.Exception());
+        return;
+      }
+      auto source_map_obj = script.ToLocalChecked()->Run(context);
```
Comment (piscisaureus) [inline src/binding.cc]: I'm a little worried that this causes both the object and the string to be embedded in the binary. Did you check?

Comment (ry) [reply src/binding.cc]: yes, I checked the binary size remains the same

Comment (piscisaureus) [reply src/binding.cc]: Ah, I see that before this patch the source map JSON was decoded in a similar way, so it wouldn't make a difference.

**Comment by piscisaureus** [comment]
Also reviewed the "Organize libdeno functions." commit now. LGTM

**Pull request merged**